src/dbus/methods/player_peer_info.c

```diff
@@ -0,0 +1,158 @@
+/**
+ * WarfaceBot, a blind XMPP client for Warface (FPS)
+ * Copyright (C) 2015-2017 Levak Borok <levak92@gmail.com>
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <gio/gio.h>
+#include <wb_dbus_methods.h>
+#include <wb_xmpp_wf.h>
+#include <wb_tools.h>
+#include <wb_xmpp.h>
+#include <wb_cvar.h>
+
+
+struct cb_args
+{
+    Warfacebot *object;
+    GDBusMethodInvocation *invocation;
+};
+
+void cmd_peer_player_info_dbus_cb(const struct player_peer_info *ppi,
+                                  void *args)
+{
+    struct cb_args *a = (struct cb_args *) args;
+
+    GVariantBuilder builder;
+    GVariant *playerinfodict;
+
+    g_variant_builder_init(&builder, G_VARIANT_TYPE("a{sv}"));
+
+    if (ppi == NULL)
+    {
+        g_variant_builder_add(&builder, "{sv}", "error", g_variant_new_uint32(-1));
```

Comment:
`uint32` and `"-1"` don't go well together.
I actually suggest to use either a boolean `true` or just error=1.
Same for the other error if player is offline.

Reply:
done